packages/components/src/number-control/types.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import type { HTMLInputTypeAttribute, InputHTMLAttributes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { InputControlProps } from '../input-control/types';
+
+export type NumberControlProps = Omit<
+	InputControlProps,
+	'isDragEnabled' | 'min' | 'max' | 'required' | 'type' | 'value'
+> & {
+	/**
+	 * If true, the default `input` HTML arrows will be hidden.
+	 *
+	 * @default false
+	 */
+	hideHTMLArrows?: boolean;
+	/**
+	 * If true, enables mouse drag gestures.
+	 *
+	 * @default true
+	 */
+	isDragEnabled?: boolean;
+	/**
+	 * If true, pressing `UP` or `DOWN` along with the `SHIFT` key will increment the
+	 * value by the `shiftStep` value.
+	 *
+	 * @default true
+	 */
+	isShiftStepEnabled?: boolean;
+	/**
+	 * The maximum `value` allowed.
+	 *
+	 * @default Infinity
+	 */
+	max?: number;
+	/**
+	 * The minimum `value` allowed.
+	 *
+	 * @default -Infinity
+	 */
+	min?: number;
+	/**
+	 * If `true` enforces a valid number within the control's min/max range.
+	 * If `false` allows an empty string as a valid value.
+	 *
+	 * @default false
+	 */
+	required?: boolean;
+	/**
+	 * Amount to increment by when the `SHIFT` key is held down. This shift value is
+	 * a multiplier to the `step` value. For example, if the `step` value is `5`,
+	 * and `shiftStep` is `10`, each jump would increment/decrement by `50`.
+	 *
+	 * @default 10
+	 */
+	shiftStep?: number;
+	/**
+	 * Amount by which the `value` is changed when incrementing/decrementing.
+	 * It is also a factor in validation as `value` must be a multiple of `step`
+	 * (offset by `min`, if specified) to be valid. Accepts the special string value `any`
+	 * that voids the validation constraint and causes stepping actions to increment/decrement by `1`.
+	 *
+	 * @default 1
+	 */
+	step?: InputHTMLAttributes< HTMLInputElement >[ 'step' ] | 'any';
```

Comment:
Mmm, this is a hard one. I'm leaning toward defining it (or *documenting it*, more like) explicitly, and in NumberControl. Mainly because the `step` prop is very relevant to common usage patterns of NumberControl specifically, so it's helpful to surface it in Docs with a good explanation.

Another minor reason is because NumberControl explicitly sets a default step value of `1` regardless of the input `type`, which is good to document because apparently the `time` type has a [built-in default of `60`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/step#syntax). Though, I don't really care about this one because NumberControl arrow key stepping doesn't really work properly with `type`s other than `number` anyway. (Why is `type` even exposed? 🤔)

Maybe another way we can go is to remove the definition and add stories with usage examples. But even with additional stories, I think I prefer to have it in the Docs props table anyway.

Reply:
Cool, makes sense